cpp/src/Ice/ObjectAdapterI.cpp

```diff
@@ -513,6 +515,17 @@ Ice::ObjectAdapterI::findServantLocator(const string& prefix) const
     return _servantManager->findServantLocator(prefix);
 }
 
+ObjectPtr
+Ice::ObjectAdapterI::dispatcher() const noexcept
+{
+    // _dispatcher is immutable, so no need to lock _mutex. There is no need to clear _dispatcher during destroy
+    // because _dispatcher does not hold onto this object adapter directly. It can hold onto a communicator that holds
+    // onto this object adapter, but the communicator will release this refcount when it is destroyed or when the
+    // object adapter is destroyed.
```

Comment:
I think much of this comment belongs next to the declaration in the header file.